addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1677,6 +1703,117 @@ export class OdooEditor extends EventTarget {
         this.observerActive('_stopContenteditable');
     }
 
+    // TABLE MANAGEMENT
+    // ================
+
+    /**
+     * Handle the selection of table cells rectangularly (as opposed to line by
+     * line from left to right then top to bottom). If such a special selection
+     * was indeed applied, return true (and false otherwise).
+     *
+     * @private
+     * @param {MouseEvent|undefined} [ev]
+     * @returns {boolean}
+     */
+    _handleSelectionInTable(ev=undefined) {
+        this.deselectTable();
+        const traversedNodes = getTraversedNodes(this.editable);
+        if (!traversedNodes.some(node => !!closestElement(node, 'td'))) {
+            // There is no table cell in the selection.
+            return false;
+        }
+        const selection = this.document.getSelection();
+        let range;
+        if (selection.rangeCount > 1) {
+            // Firefox selection in table works differently.
+            const startRange = getDeepRange(this.editable, {range: selection.getRangeAt(0)});
+            const endRange = getDeepRange(this.editable, {range: selection.getRangeAt(selection.rangeCount - 1)});
+            range = this.document.createRange();
+            range.setStart(startRange.startContainer, 0);
+            range.setEnd(endRange.startContainer, 0);
+        } else {
+            range = getDeepRange(this.editable);
+        }
+        const startTd = closestElement(range.startContainer, 'td');
+        const endTd = closestElement(range.endContainer, 'td');
+        let appliedCustomSelection = false;
+        const startTable = closestElement(range.startContainer, 'table');
+        const endTable = closestElement(range.endContainer, 'table');
+        if (startTd !== endTd && startTable === endTable) {
+            // The selection goes through at least two different cells -> select
+            // cells.
+            this._selectTableCells(range);
+            appliedCustomSelection = true;
+        } else if (!traversedNodes.every(node => !!closestElement(node, 'td'))) {
+            // The selection goes through a table but also outside of it ->
+            // select the whole table.
+            for (const table of new Set(traversedNodes.map(node => closestElement(node, 'table')))) {
+                if (table) {
+                    table.classList.toggle('o_selected_table', true);
+                    for (const td of table.querySelectorAll('td')) {
+                        td.classList.toggle('o_selected_td', true);
+                    }
```

Comment:
Because we're selecting the full table, which requires none of the computations done in `_selectTableCells`.
This is a special case we can observe in Google Docs and others: the individual cell selection thing is only done when we're selecting _within_ a table ; if we select anything more, going through the table selects the whole table.